client/src/pages/AgencyHomePage/AgencyHomePageWrapper.component.tsx

```diff
@@ -0,0 +1,13 @@
+import { HomePageProvider } from '../../contexts/HomePageContext'
+
+import AgencyHomePage from './AgencyHomePage.component'
+
+const AgencyHomePageWrapper = (): JSX.Element => {
+  return (
+    <HomePageProvider>
+      <AgencyHomePage />
+    </HomePageProvider>
+  )
+}
+
+export default AgencyHomePageWrapper
```

Comment:
This is a bit of a code smell to me, since it breaks consistency with how HomePage is structured. Not entirely your fault though, since AgencyHomePage is quite bloated. A further issue should extract the hook-dependent components into a separate file, and merge the wrapper into AgencyHomePage.

Reply:
Agree 100%!